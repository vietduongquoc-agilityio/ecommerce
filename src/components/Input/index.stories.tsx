import { Meta, StoryObj } from "@storybook/react";
import Input from "./";

export default {
  title: "Components/Input",
  component: Input,
  args: {
    placeholder: "Enter text...",
    className: "",
  },
  argTypes: {
    typeInput: {
      control: "radio",
      options: ["text", "number"],
    },
  },
} as Meta<typeof Input>;

export const TextInput: StoryObj<typeof Input> = {
  args: {
    typeInput: "text",
  },
};

export const NumberInput: StoryObj<typeof Input> = {
  args: {
    typeInput: "number",
    placeholder: "Enter number...",
  },
};
