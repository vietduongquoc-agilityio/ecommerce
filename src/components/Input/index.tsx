import * as React from "react";
import { Slot } from "@radix-ui/react-slot";

export type InputType = "text" | "number";

interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {
  typeInput?: InputType;
  asChild?: boolean;
}

const Input = React.forwardRef<HTMLInputElement, InputProps>(
  ({ typeInput = "text", asChild, className, ...props }, ref) => {
    const Comp = asChild ? Slot : "input";

    return (
      <Comp
        ref={ref}
        type={typeInput}
        className={`border rounded px-3 py-2 text-sm focus:outline-none focus:ring-2 focus:ring-black ${className}`}
        {...props}
      />
    );
  }
);

Input.displayName = "Input";

export default Input;
