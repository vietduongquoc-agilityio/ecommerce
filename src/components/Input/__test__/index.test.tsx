import React from "react";
import { render } from "@testing-library/react";
import Input from "../";

describe("Input Snapshot Test", () => {
  it("renders text input correctly", () => {
    const { asFragment } = render(<Input typeInput="text" placeholder="Enter text" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders number input correctly", () => {
    const { asFragment } = render(<Input typeInput="number" placeholder="Enter number" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
