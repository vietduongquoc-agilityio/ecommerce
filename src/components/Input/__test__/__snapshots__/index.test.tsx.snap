// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Snapshot Test renders number input correctly 1`] = `
<DocumentFragment>
  <input
    class="border rounded px-3 py-2 text-sm focus:outline-none focus:ring-2 focus:ring-black undefined"
    placeholder="Enter number"
    type="number"
  />
</DocumentFragment>
`;

exports[`Input Snapshot Test renders text input correctly 1`] = `
<DocumentFragment>
  <input
    class="border rounded px-3 py-2 text-sm focus:outline-none focus:ring-2 focus:ring-black undefined"
    placeholder="Enter text"
    type="text"
  />
</DocumentFragment>
`;
